### DISCUSSION TITLE
The License Question

### DISCUSSION THREAD

**Discussion by jakubmanczak** [debatecore/.github#5] [General]
Our work on the software we create to aid public debate is done in the open, with our commit histories available to all and our _code_ open to being surveyed and reviewed. The Debate Tools project has, on the front page, a note of being available freely.

Yet, none of the debatecore repositories have a license. It's not mandatory to have one, of course, but I feel we should sort this out sooner rather than later, to avoid future headaches. I understand that we generally want to have the projects remain that way, but the specific flavour of open-source license we pick is still up in the air.

Obviously, any broader articles or written works under the umbrella of debatecore would not be released so permissively. Their only uses are to be read and cited, not shared or edited – and this can and should happen without licensing. The license question thus only pertains to our code and technologies.

And then there's the copyright question.
The Debate Tools project, for instance, was at first entirely a creation of mine. A notice of copyright being held by me was justified, natural and obvious. But now, with the creation of debatecore (the ultimate shape of which is still undecided, causing a need for further clarification), should the copyright be transferred somehow? If debatecore doesn't have a legal personality, as it's not a legally registered institution, then I don't think it can "hold" it for us. A license needs to be signed by some entity, to whom the licensed work belongs.

Which license to pick is a question of semantics, conventions and priorities. Most licenses share the source code and allow anyone to edit it and experiment with it. Most allow for both personal and commercial use, and most denote a lack of a warranty and any liability from the authors.

Where licenses differ is in a few categories. Most notably, whether to require contribution back to the main project or the prohibition of closed source derivative works, which is what happens under the GPLv3 license and similar ones, or which doesn't happen in others, like the ISC license.

But the ISC license is a lesser known one. In fact, The Publications Office of the European Union advises the use of the MIT license instead, to avoid [license proliferation](https://op.europa.eu/en/web/eu-vocabularies/concept/-/resource?uri=http://publications.europa.eu/resource/authority/licence/ISC). Even ISC themselves abandoned it. But it's an example of an MIT-like license that omits all the wording "unnecessarily included after the [Berne Convention](https://en.wikipedia.org/wiki/Berne_Convention)".

It's a small issue, but it's in my blood to overcomplicate things that need not be dwelled on so much. Let's!

~jm

**Comment by Mateusz-Dobrzynski** [comment]
Okay, let's break it down.

1. **Copyright**. I've done some research, and no, debatecore cannot own copyright, unless we turn it into a foundation or any other organization with legal personality (which, as for now, I don't deem necessary). Who owns the copyright, then? Since no agreements or contracts were signed, it seems that everyone owns their respective works (this probably needs further research). But the "© Jakub Mańczak" clause regards the website, as seen by the user, not the source code. A sticking plaster solution would probably be to extend the clause to "© Jakub Mańczak & Mateusz Dobrzyński", as we are the main project creators. Where do we draw the line with other contributors, though? That's why we need a license for the source code – if you contribute to a project, you should know your work can be reproduced and reused in the future. This is what Wikipedia does. It constantly reminds you that by contributing, you waive any copyright. And it works, despite no legally binding agreements being signed.
2. **License**. We should ask ourselves what we want the license to permit and find one that fits us:
  - Commercial use – sure
  - Reproducing, changing, resuing – sure
  - Liability, warranty – of course not
  I'd also consider these (which would exclude MIT from our options):
  - Same terms (closed source distributions prohibition)?
  - Attribution required?

**Comment by jakubmanczak** [comment]
There's a few licenses fitting our requirements here, and [GNU AGPLv3](https://choosealicense.com/licenses/agpl-3.0/) seems like the best pick. It's not like anyone is going to be building software on top of our projects, so the clause of attribution isn't going to become a burden for anyone, and it should theoretically protect the acknowledgement of our contribution to the debate scene if anyone chooses to expand on our work as a base were we to abandon it.

I'll examine it closer before going forward.

**Comment by jakubmanczak** [comment]
A pull request with the appropriate license has been added to [debate tools](https://github.com/debatecore/debate-tools). Closing.
